porch/api/porch/types_packagerevisions.go

```diff
@@ -40,14 +40,22 @@ type PackageRevisionList struct {
 	Items []PackageRevision
 }
 
-// PackageRevisionSpec defines the desired state of PackageRevision
-type PackageRevisionSpec struct {
-	PackageName string `json:"packageName,omitempty"`
+type PackageRevisionType string
 
-	Revision string `json:"revision,omitempty"`
+const (
+	PackageRevisionTypeDraft    PackageRevisionType = "draft"
+	PackageRevisionTypeProposed PackageRevisionType = "proposed"
+	PackageRevisionTypeFinal    PackageRevisionType = "final"
+)
 
+// PackageRevisionSpec defines the desired state of PackageRevision
+type PackageRevisionSpec struct {
+	PackageName    string `json:"packageName,omitempty"`
+	Revision       string `json:"revision,omitempty"`
 	RepositoryName string `json:"repository,omitempty"`
 
+	Type PackageRevisionType `json:"type,omitempty"`
```

Comment:
Let's not change it in this PR (I assume it'll break the whole chain), but this reads more like `state` or `lifecycle` than `type` to me.

Reply:
I decided to do this in this PR ... there will be potential for less long-term fallout. Thanks!